### DISCUSSION TITLE
AbpCrudPageBase + Blazorise DataGrid + Filtering Question

### DISCUSSION THREAD

**Discussion by 274188A** [abpframework/abp#6381] [Q&A] [answered]
Looking for some guidance on how a developer utilizing the [AbpCrudPageBase](https://github.com/abpframework/abp/blob/dev/framework/src/Volo.Abp.BlazoriseUI/AbpCrudPageBase.cs) should go about hooking into the [Blazorise DataGrid Filter](https://blazorise.com/docs/extensions/datagrid/#filtering) functionality.

Maybe something @stsrki can assist with? I've not seen any samples demonstrating this so far.

The blazorise doco (extract below) seems to assume we have the entire data-set loaded onto the client begin with:

>*Custom Filtering*
Filter API is fairly straightforward. All you need is to attach CustomFilter to a function and bind search value to TextEdit field. DataGrid will automatically respond to entered value.

I'm not sure this is going to work with the AbpCrudPageBase where we are paging

Thanks, 
John

**Comment by 274188A** [comment, marked as answer]
I think I see where my confusion comes from: I am using Custom Filtering - but I assumed that by setting Filterable="true" on a column that i would then be able to hook into the changed event of the filter text boxes (the ones that get auto created for you) that appear above the columns for filtering. 

Seems I have to use my _own_ filter textboxes for custom filtering. in fact custom filtering is ONLY usable for in memory :-(

**Comment by stsrki** [comment]
Having data in memory is one way. The other is to hook to the outside API through the `ReadData` handler. You can find it in the documentation https://blazorise.com/docs/extensions/datagrid/#large-data-example

In `ReadData` mode you will get everything regarding the filtering through its `eventArgs`. 

CustomFilter is a little different because if operated "outside" of DataGrid and whatever you use with it is totally upon yourself. After you do the filtering with `CustomFilter` you then just re-apply the filtered data to the DataGrid.

**Comment by 274188A** [comment]
Yes - the AbpCrudPageBase uses the ReadData mode - I have overriden a method `UpdateGetListInputAsync` to set my filtering.  Thanks for  the help!
